Fifth homework/JS_HW_5.js

```diff
@@ -0,0 +1,101 @@
+const pr = (n) => console.log(n);
+const line = () => console.log(" \n======================CONTINUE======================\n ");
+
+/* 1. Бесконечные аргументы
+  - Напишите функцию, принимающую на вход любое количество массивов
+  - Функция возвращает массив, в котором будут находиться все переданные в функцию в виде чисел
+  - Например: mergeArrays([1,2], [3,4], [5,6]) // [1,2,3,4,5,6]
+  - Решить с использованием Spread operator */
+
+  function mergeArrays(...arrFunc) { // раз спред - всегда спред)
+    const numArray = []; // в новый массив будем скаладывать наши входящие массивы
+    for (const arr1 of arrFunc) numArray.push(...arr1); //перебираем входящий массив с массивами и пушим спредом в наш созданный. решил попробывать соркащенное написание))
+    return numArray;
+  }
+  pr(mergeArrays([1, 2], [3, 4], [5, 6]));
+  pr(mergeArrays([1, 2], [3, 4], [5, 6],[7, 8], [9, 10], [11, 12]));
+  line();
+
+/* 2. Devide by _
+  - Написать функцию, которая преобразует любое предложение в вот_Такой_Вот_Вид и возвращает его. 
+  - Первое слово должно начинаться с буквы в нижнем регистре, у остальных -  верхнем. 
+  - Пример: I am super engineer => i_Am_Super_Engineer */
+
+  function devideBy(a = "") { // сразу переменную в текст
+    let textEdit = a.replaceAll(" ", "_").toLowerCase(); // меняем старое на новое и всё в низ - условие с первым словом в кармане)
+    for (i = 0; i < textEdit.length; i++)
+       if (textEdit[i - 1] === "_") textEdit[i].toUpperCase(); // опять сокращаю) по "_" хитрим и меняем символы вверх.
+    return textEdit;
+  }
+  pr(devideBy("ПривеТ! Я коТлеТка с МЯСом)")); // текст диктовал сын))
+  pr(devideBy("I am super engineer"));
+  line();
+
+// /* 3. Фибаначчи
+//   - Напишите функцию fibanacci(n), возвращающую энное число Фибоначчи
+//   - числа Фибоначчи (строка Фибоначчи) — числовая последовательность, первые два числа которой являются 0 и 1, а каждое последующее за ними число является 
+//   суммой двух предыдущих
+//   - Например fibanacci(8) //21 */
+
+function fibanacci(n) {
```

Comment:
Можно вьінести условие и сразу возвращать значение не заходя в цикл
А массив сделать [0, 1]
```suggestion
function fibanacci(n) {
// i == 0 || i == 1
```